extensions/wikia/CreateNewWiki/tasks/ConfigureWikiFactory.php

```diff
@@ -0,0 +1,262 @@
+			: "";
+
+		while ( $isExist == false ) {
+			$dirName = self::IMGROOT . $prefix . "/" . $dir_base . $suffix . $dir_lang . "/images";
+
+			if ( self::wgUploadDirectoryExists( $dirName ) ) {
+				$suffix = rand( 1, 9999 );
+			} else {
+				$dir_base = $dir_base . $suffix;
+				$isExist = true;
+			}
+		}
+
+		$this->debug( implode( ":", [ "CreateWiki", __METHOD__, "Returning '{$dir_base}'" ] ) );
+		wfProfileOut( __METHOD__ );
+		return $dir_base;
+	}
+
+	public static function wgUploadDirectoryExists( $sDirectoryName ) {
+		wfProfileIn( __METHOD__ );
+		$iVarId = WikiFactory::getVarIdByName( 'wgUploadDirectory' );
+
+		// Crash immediately if $iVarId is not a positive integer!
+		\Wikia\Util\Assert::true( $iVarId );
+
+		$aCityIds = WikiFactory::getCityIDsFromVarValue( $iVarId, $sDirectoryName, '=' );
+		wfProfileOut( __METHOD__ );
+		return !empty($aCityIds);
+	}
+
+	/**
+	 * Sanitizes a name to be a valid S3 bucket name. It means it can contain only letters and numbers
+	 * and optionally hyphens in the middle. Maximum length is 63 characters, we're trimming it to 55
+	 * characters here as some random suffix may be added to solve duplicates.
+	 *
+	 * Note that different arguments may lead to the same results so the conflicts need to be solved
+	 * at a later stage of processing.
+	 *
+	 * @see http://docs.aws.amazon.com/AmazonS3/latest/dev/BucketRestrictions.html
+	 *      Wikia change: We accept underscores wherever hyphens are allowed.
+	 *
+	 * @param $name string Directory name
+	 * @return string Sanitized name
+	 */
+	private static function sanitizeS3BucketName( $name ) {
+		if ( $name == 'admin' ) {
+			$name .= 'x';
+		}
+
+		$RE_VALID = "/^[a-z0-9](?:[-_a-z0-9]{0,53}[a-z0-9])?(?:[a-z0-9](?:\\.[-_a-z0-9]{0,53}[a-z0-9])?)*\$/";
+		# check if it's already valid
+		$name = mb_strtolower( $name );
+		if ( preg_match( $RE_VALID, $name ) && strlen( $name ) <= self::SANITIZED_BUCKET_NAME_MAXIMUM_LENGTH ) {
+			return $name;
+		}
+
+		# try fixing the simplest and most popular cases
+		$check_name = str_replace( [ '.', ' ', '(', ')' ], '_', $name );
+		if ( in_array( substr( $check_name, -1 ), [ '-', '_' ] ) ) {
+			$check_name .= '0';
+		}
+		if ( preg_match( $RE_VALID, $check_name ) && strlen( $check_name ) <= self::SANITIZED_BUCKET_NAME_MAXIMUM_LENGTH ) {
+			return $check_name;
+		}
+
+		# replace invalid ASCII characters with their hex values
+		$s = '';
+		for ( $i = 0; $i < strlen( $name ); $i++ ) {
+			$c = $name[$i];
+			if ( $c >= 'a' && $c <= 'z' || $c >= '0' && $c <= '9' ) {
+				$s .= $c;
+			} else {
+				$s .= bin2hex( $c );
+			}
+			if ( strlen( $s ) >= self::SANITIZED_BUCKET_NAME_MAXIMUM_LENGTH ) {
+				break;
+			}
+		}
+		$name = substr( $s, 0, self::SANITIZED_BUCKET_NAME_MAXIMUM_LENGTH );
+
+		return $name;
+	}
+
+	public function getImagesURL( $language ) {
+
+		$imagesURL = $this->prepareDirValue( $this->taskContext->getWikiName(), $language );
+
+		if ( isset($language) && $language !== "en" ) {
```

Comment:
We seem to be using "en" everywhere, it would be good to use some global const instead (not sure if we have one)